### PR TITLE
log metrics

### PR DESCRIPTION
**Pull request by sowu880**

### CODE REVIEW

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Common/Metrics/SuccessfulDataSizeMetric.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+
+namespace Microsoft.Health.Fhir.Synapse.Common.Metrics
+{
+    public class SuccessfulDataSizeMetric : DiagnosticMetrics
```
Comment (tongwu-msft) [inline FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Common/Metrics/SuccessfulDataSizeMetric.cs:10]: DimensionNames.IsDiagnostic, false? Typo?

**Review by tongwu-msft — APPROVED** [review]

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.HealthCheck/HealthCheckBackgroundService.cs [hunk 2]
```diff
@@ -53,6 +57,7 @@ protected override async Task ExecuteAsync(CancellationToken cancellationToken)
 
                     // Todo: Send notification to mediator and remove listeners here.
                     var listenerTasks = _healthCheckListeners.Select(l => l.ProcessHealthStatusAsync(healthStatus, cancellationToken)).ToList();
+                    _metricsLogger.LogHealthStatusMetric(healthStatus.Status == Models.HealthCheckStatus.HEALTHY ? 1 : 0);
```
Comment (QuanWanxx) [inline FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.HealthCheck/HealthCheckBackgroundService.cs:60]: Questions: The 1 indicate health and 0 is unhealth? The metric logs also contain numbers for status?

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Common/Metrics/MetricsLogger.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+            _logger = logger;
+        }
+
+        public void LogMetrics(Metrics metrics, double metricsValue)
+        {
+            _logger.LogInformation(
+                $"Log Metrics \r\n " +
+                $"{metrics.Name} : {metricsValue} \r\n " +
+                $"Dimensions: \r\n" +
+                $"{string.Join("\r\n", metrics.Dimensions.Select(x => string.Join(":", x.Key, x.Value.ToString())))}");
```
[the remark this answers is not in the document]
Comment (sowu880) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Common/Metrics/MetricsLogger.cs]: Thanks, fixed it.

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Common/Metrics/MetricsLoggerExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        {
+            metricsLogger.LogMetrics(new ProcessLatencyMetric(), value);
+        }
+
+        public static void LogHealthStatusMetric(this IMetricsLogger metricsLogger, double value)
+        {
+            metricsLogger.LogMetrics(new HealthStatusMetric(), value);
+        }
+
+        public static IServiceCollection AddMetricsLogger(this IServiceCollection services)
```
[the remark this answers is not in the document]
Comment (sowu880) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Common/Metrics/MetricsLoggerExtensions.cs]: Added new metrics logger registration class.

**Review by QuanWanxx — APPROVED** [review]

**Pull request merged**